functions/Remove-DbaCredential.ps1

```diff
@@ -0,0 +1,133 @@
+
+    .PARAMETER SqlCredential
+        Login to the target instance using alternative credentials. Accepts PowerShell credentials (Get-Credential).
+        Windows Authentication, SQL Server Authentication, Active Directory - Password, and Active Directory - Integrated are all supported.
+        For MFA support, please use Connect-DbaInstance.
+
+    .PARAMETER Credential
+        Specifies one or more SQL credential(s) to delete. If unspecified, all accounts will be removed.
+
+    .PARAMETER ExcludeCredential
+        Specifies one or more SQL credential(s) to exclude.
+
+    .PARAMETER Identity
+        Only include specific identities
+        Note: if spaces exist in the credential identity, you will have to type "" or '' around it.
+
+    .PARAMETER ExcludeIdentity
+        Excluded identities
+
+    .PARAMETER InputObject
+        Allows piping from Get-DbaCredential.
+
+    .PARAMETER WhatIf
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+        This is the default. Use -Confirm:$false to suppress these prompts.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Credential
+        Author: Mikey Bronowski (@MikeyBronowski), https://bronowski.it
+        dbatools PowerShell module (https://dbatools.io)
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Remove-DbaCredential
+
+    .EXAMPLE
+        PS C:\> Remove-DbaCredential -SqlInstance localhost, localhost\namedinstance
+
+        Removes all SQL credentials on the localhost, localhost\namedinstance instances.
+
+    .EXAMPLE
+        PS C:\> Remove-DbaCredential -SqlInstance localhost -Credential MyDatabaseCredential
+
+        Removes MyDatabaseCredential SQL credential on the localhost.
+
+    .EXAMPLE
+        PS C:\> Get-DbaCredential -SqlInstance SRV1 | Out-GridView -Title 'Select SQL credential(s) to drop' -OutputMode Multiple | Remove-DbaCredential
+
+        Using a pipeline this command gets all SQL credentials on SRV1, lets the user select those to remove and then removes the selected SQL credentials.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [Parameter(ParameterSetName = 'NonPipeline', Mandatory = $true, Position = 0)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [Parameter(ParameterSetName = 'NonPipeline')]
```

Comment:
Instead of NonPipeline, you can add a DefaultParameterSetName as seen in our copy commands. Makes it much prettier.

`[cmdletbinding(DefaultParameterSetName = "Default", SupportsShouldProcess, ConfirmImpact = "High")]`

Reply:
Do you mean removing `[Parameter(ParameterSetName = 'NonPipeline')]` completely after that?

I also found some commands that are missing the `DefaultParameterSetName = "Default"` (including Copy-* commands)
`[cmdletbinding(SupportsShouldProcess, ConfirmImpact = "Medium")]`

Copy-AgentSchedule for example...

I can also open another issue updating the missing DefaultParameterSetName for the previous commands if that helps?